src/xtopcom/xdata/src/xrelay_block.cpp

```diff
@@ -403,24 +403,46 @@ void xrelay_block::set_block_merkle_root_hash(evm_common::h256 hash)
     m_header.set_block_merkle_root_hash(hash);
 }
 
-void xrelay_block::make_merkle_root_hash(const std::vector<evm_common::h256>& hash_vector)
+void xrelay_block::set_tx_blocks_info_and_make_block_merkle_root(const std::vector<uint64_t>& height_vector, const std::vector<evm_common::h256>& hash_vector)
```

Comment:
does this API contains too much functionalities? set is set and make merkle root is make merkle root.